admin/requirements-dev.txt

```diff
@@ -178,13 +178,17 @@ six==1.11.0 \
 tox==2.9.1 \
     --hash=sha256:752f5ec561c6c08c5ecb167d3b20f4f4ffc158c0ab78855701a75f5cef05f4b8 \
     --hash=sha256:8af30fd835a11f3ff8e95176ccba5a4e60779df4d96a9dfefa1a1704af263225
-urllib3==1.23 \
-    --hash=sha256:a68ac5e15e76e7e5dd2b8f94007233e01effe3e50e8daddf69acfd81cb686baf \
-    --hash=sha256:b5725a0bd4ba422ab0e66e89e030c806576753ea3ee08554382c14e685d117b5
+urllib3==1.25.3 \
+    --hash=sha256:b246607a25ac80bedac05c6f282e3cdaf3afb65420fd024ac94435cabe6e18d1 \
+    --hash=sha256:dbe59173209418ae49d485b87d1681aefa36252ee85884c31346debd19463232
 virtualenv==15.1.0 \
     --hash=sha256:02f8102c2436bb03b3ee6dede1919d1dac8a427541652e5ec95171ec8adbc93a \
     --hash=sha256:39d88b533b422825d644087a21e78c45cf5af0ef7a99a1fc9fbb7b481e5c85b0 \
     # via tox
 wrapt==1.10.11 \
     --hash=sha256:d4d560d479f2c21e1b5443bbd15fe7ec4b37fe7e53d335d3b9b0a7b1226fe3c6 \
     # via astroid
+
+# WARNING: The following packages were not pinned, but pip requires them to be
```

Comment:
Any idea why this warning is here? Is it due to the new version of pip-tools?

Reply:
[Yes](https://github.com/jazzband/pip-tools/issues/806). Since `pip-compile` refuses to hash `setuptools` unless given `--allow-unsafe`, if whatever version of `setuptools` was installed when the virtualenv was created needed to be upgraded to satisfy another package's dependencies, `pip` would fail.

There's some discussion on that issue suggesting that it would be safer to use `--allow-unsafe` to let `pip-compile` pin and hash `setuptools`. That makes sense to me, but that should probably be a separate PR, and maybe @kushaldas would like to weigh in here.